### PR TITLE
feat: Add basic metadata info to new behavior guide.

### PR DESCRIPTION
**Pull request by petejohanson**

### CODE REVIEW

File: docs/docs/development/new-behavior.mdx [hunk 1]
```diff
@@ -301,6 +303,97 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+
+Behavior metadata consists of one or more metadata sets, where each metadata set has a set of values for the parameter(s) used with the behavior.
+
+For example, a common approach for behaviors is to have a set of possible first parameters that identify the "command" to invoke for the behavior, and the second parameter is a detail/sub-parameter to the action. You can see this with the `&bt` behavior.
+In that scenario, all `&bt` "commands" that take a BT profile as a secound parameter are grouped into one set, and all commands that take no arguments are grouped into another.
+
+This allows the ZMK Studio UI to properly show a input for a profile only when the appropriate first "command" selection is made in the UI. Here is a snippet of that setup from the [behavior_bt.c](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_bt.c#L25) code:
+
+```
+
```
Comment (joelspadin) [inline docs/docs/development/new-behavior.mdx:325]:
```suggestion
```c
```

File: docs/docs/development/new-behavior.mdx [hunk 2]
```diff
@@ -301,6 +303,97 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+# Set up the metadata set for the commands that take a profile for the second parameter.
+static const struct behavior_parameter_metadata_set profile_index_metadata_set = {
+    .param1_values = prof_index_param1_values,
+    .param1_values_len = ARRAY_SIZE(prof_index_param1_values),
+    .param2_values = prof_index_param2_values,
+    .param2_values_len = ARRAY_SIZE(prof_index_param2_values),
+};
+
+# Finally, expose all the sets in the top level aggregate structure.
```
Comment (joelspadin) [inline docs/docs/development/new-behavior.mdx:387]: This doesn't seem to be the final part that's necessary. Shouldn't we show this getting set in the behavior_driver_api struct?

**Review by joelspadin — CHANGES_REQUESTED** [review]

File: docs/docs/development/new-behavior.mdx (hunk 2 above)
Comment (petejohanson) [reply docs/docs/development/new-behavior.mdx:388]: Indeed. Added.

File: docs/docs/development/new-behavior.mdx [hunk 3]
```diff
@@ -301,6 +303,114 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+
+static const struct behavior_driver_api behavior_bt_driver_api = {
+    .binding_pressed = on_keymap_binding_pressed,
+    .binding_released = on_keymap_binding_released,
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+    .parameter_metadata = &metadata,
+#endif // IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+};
+
+
```
Comment (caksoylar) [inline docs/docs/development/new-behavior.mdx:411]:
```suggestion
```

File: docs/docs/development/new-behavior.mdx [hunk 4]
```diff
@@ -301,6 +303,114 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
     For unibody keyboards, all locality values perform the same as `BEHAVIOR_LOCALITY_GLOBAL`.
     :::
 
+##### Behavior Metadata
+
+allows flexibility to specify different kinds of well known parameter types, such as a HID usage, different second parameters passed on the selected first parameter, etc.
+Behavior metadata documents the possible combinations of parameters that can be used with the behavior when added to your keymap. The metadata structure
```
Comment (Nick-Munnich) [inline docs/docs/development/new-behavior.mdx:309]:
```suggestion
Behavior metadata documents the possible combinations of parameters that can be used with the behavior when added to your keymap. The metadata structure allows flexibility to specify different kinds of well known parameter types, such as a HID usage, different second parameters passed on the selected first parameter, etc.
```
This looks like an ooopsie.

File: docs/docs/development/new-behavior.mdx [hunk 5]
```diff
@@ -301,6 +303,114 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+
+You can see a few examples of how the metadata is implemented in practice for:
+
+- [key press](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_key_press.c#L21),
+- [RGB underglow](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_rgb_underglow.c#L23),
+- and more complex scenarios like [hold-tap](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_hold_tap.c#L680) which is dynamic based on what behaviors are set up in the hold-tap bindings
```
Comment (Nick-Munnich) [inline docs/docs/development/new-behavior.mdx:315]:
```suggestion
- [Key press](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_key_press.c#L21)
- [RGB underglow](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_rgb_underglow.c#L23)
- [Hold-tap](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_hold_tap.c#L680), which is dynamic based on what behaviors are set up in the hold-tap bindings
```
I think this bit is nicer when more succinct.

File: docs/docs/development/new-behavior.mdx [hunk 6]
```diff
@@ -301,6 +303,114 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+
+Behavior metadata consists of one or more metadata sets, where each metadata set has a set of values for the parameter(s) used with the behavior.
+
+For example, a common approach for behaviors is to have a set of possible first parameters that identify the "command" to invoke for the behavior, and the second parameter is a detail/sub-parameter to the action. You can see this with the `&bt` behavior.
+In that scenario, all `&bt` "commands" that take a BT profile as a secound parameter are grouped into one set, and all commands that take no arguments are grouped into another.
```
Comment (Nick-Munnich) [inline docs/docs/development/new-behavior.mdx:320]:
```suggestion
In that scenario, all `&bt` "commands" that take a BT profile as a second parameter are grouped into one set, and all commands that take no arguments are grouped into another.
```

File: docs/docs/development/new-behavior.mdx [hunk 7]
```diff
@@ -301,6 +303,114 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
...
+
+This allows the ZMK Studio UI to properly show a input for a profile only when the appropriate first "command" selection is made in the UI. Here is a snippet of that setup from the [behavior_bt.c](https://github.com/zmkfirmware/zmk/blob/main/app/src/behaviors/behavior_bt.c#L25) code:
+
+```c
+
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+# Set up the values for commands that take no additional parameter.
```
[the remark this answers is not in the document]
Comment (petejohanson) [reply docs/docs/development/new-behavior.mdx]: Done.

**Review by petejohanson — COMMENTED** [review]
Addressed everyone's code review. Thanks!

File: docs/docs/development/new-behavior.mdx [hunk 8]
```diff
@@ -301,6 +303,111 @@ Comes in the form `static const struct behavior_driver_api <behavior_name>_drive
     For unibody keyboards, all locality values perform the same as `BEHAVIOR_LOCALITY_GLOBAL`.
     :::
 
+##### Behavior Metadata
```
Comment (caksoylar) [inline docs/docs/development/new-behavior.mdx:306]:
```suggestion
##### Behavior metadata
```

Nit, we use [title case](https://zmk.dev/docs/development/contributing/documentation#formatting-and-linting-your-changes) for heading levels 1-3.

**Review by caksoylar — APPROVED** [review]
LGTM.

**Pull request merged**